components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/cache/AuthenticationContextCache.java

```diff
@@ -97,8 +105,29 @@ public void addToCache(AuthenticationContextCacheKey key, AuthenticationContextC
                             ", Operation : STORE ]";
                     log.debug("Authentication context is stored with details " + message);
                 }
+                if (isSessionDataStorageOptimizationEnabled && entry.getContext() != null) {
+                    try {
+                        AuthenticationContextLoader.getInstance().optimizeAuthenticationContext(entry.getContext());
+                    } catch (SessionDataStorageOptimizationException e) {
+                        if (log.isDebugEnabled()) {
+                            log.debug(String.format("Error occurred while optimizing the Authentication context with " +
+                                    "context id: %s", entry.getContext().getContextIdentifier()), e);
+                        }
+                        return;
+                    }
+                }
                 SessionDataStore.getInstance().storeSessionData(key.getContextId(), AUTHENTICATION_CONTEXT_CACHE_NAME,
                         entry, tenantId);
+                if (isSessionDataStorageOptimizationEnabled) {
```

Comment:
Lets remove this if condition

Reply:
Fixed with [`c43c465`](https://github.com/wso2/carbon-identity-framework/pull/4446/commits/c43c465fb4b8edc33bf309a076d8e8226b5df9e1)